Loop/View Models/SimpleBolusViewModel.swift

```diff
@@ -0,0 +1,315 @@
+                        break
+                    }
+                }
+            } else {
+                completion()
+            }
+        }
+        
+        func saveManualGlucose(_ completion: @escaping () -> Void) {
+            if let glucose = glucose {
+                let manualGlucoseSample = NewGlucoseSample(date: saveDate, quantity: glucose, isDisplayOnly: false, wasUserEntered: true, syncIdentifier: UUID().uuidString)
+                delegate.addGlucose([manualGlucoseSample]) { error in
+                    DispatchQueue.main.async {
+                        if let error = error {
+                            self.presentAlert(.manualGlucoseEntryPersistenceFailure)
+                            self.log.error("Failed to add manual glucose entry: %{public}@", String(describing: error))
+                        } else {
+                            completion()
+                        }
+                    }
+                }
+            } else {
+                completion()
+            }
+        }
+        
+        func saveCarbs(_ completion: @escaping () -> Void) {
+            if let carbs = carbs {
+                
+                let interaction = INInteraction(intent: NewCarbEntryIntent(), response: nil)
+                interaction.donate { [weak self] (error) in
+                    if let error = error {
+                        self?.log.error("Failed to donate intent: %{public}@", String(describing: error))
+                    }
+                }
+
+                let carbEntry = NewCarbEntry(date: saveDate, quantity: carbs, startDate: saveDate, foodType: nil, absorptionTime: nil)
+                delegate.addCarbEntry(carbEntry) { error in
+                    DispatchQueue.main.async {
+                        if let error = error {
+                            self.presentAlert(.carbEntryPersistenceFailure)
+                            self.log.error("Failed to add carb entry: %{public}@", String(describing: error))
+                        } else {
+                            completion()
+                        }
+                    }
+                }
+            } else {
+                completion()
+            }
+        }
+        
+        func enactBolus() {
+            if let bolusVolume = bolus?.doubleValue(for: .internationalUnit()), bolusVolume > 0 {
+                delegate.enactBolus(units: bolusVolume, at: saveDate)
+            }
+            didSave()
+        }
+        
+        authenticateIfNeeded {
+            saveManualGlucose {
+                saveCarbs {
+                    enactBolus()
+                }
+            }
+        }
+    }
+    
+    private func presentAlert(_ alert: Alert) {
+        dispatchPrecondition(condition: .onQueue(.main))
+
+        // As of iOS 13.6 / Xcode 11.6, swapping out an alert while one is active crashes SwiftUI.
+        guard activeAlert == nil else {
+            return
+        }
+
+        activeAlert = alert
```

Comment:
Where is `activeAlert` cleared?

Reply:
`activeAlert` is cleared via the binding passed to `.alert(...)` in the view.